src/NHibernate/Async/Type/EntityType.cs

```diff
@@ -218,12 +218,19 @@ public override Task<object> ResolveIdentifierAsync(object value, ISessionImplem
 
 			//TODO: implement caching?! proxies?!
 
+			var keyType = GetIdentifierOrUniqueKeyType(factory)
+				// EntityUniqueKey was doing this on the type. I suspect this was needed only for its usage in Loader,
+				// which can work with entities as keys not yet instanciated and just represented by their identifiers.
+				// But since removing this call from EntityUniqueKey is done for a patch and that the code path here has
+				// no known bugs with this GetSemiResolvedType, moving its call here for avoiding altering this code
+				// path. See GH1645.
+				.GetSemiResolvedType(factory);
```

Comment:
Based on the implementation, this actually could happen only if the primary or unique key is entity itself. Is it event possible?

Reply:
Yes, it's possible.